File: pgm/ABC_TypeScript/ABC_188_A.ts
import fs = require("fs");

const No: string = "No";
const Yes: string = "Yes";

class Solver
{
    arr: string[];
    line: string[];

    X: number;
    Y: number;
    input: string;

    constructor()
    {
        this.input = fs.readFileSync("/dev/stdin", "utf8");
        this.line = this.input.split("\n");
        this.arr = this.line[0].split(" ");
        this.X = parseInt(this.arr[0], 10);
        this.Y = parseInt(this.arr[1], 10);
    }

    solve()
    {
        console.log(
            Math.max(this.X, this.Y) < Math.min(this.X, this.Y) + 3 ? Yes : No);
        return 0;
    }
}

const s = new Solver();
s.solve();

File: pgm/ABC_Dart/ABC_188_A.dart
// import 'dart:collection';
// import 'dart:convert';
import 'dart:io';
import 'dart:math';
// import 'dart:typed_data';

void main() {
  final [int X, int Y] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const String Yes = 'Yes';
  const String No = 'No';
  const int three = 3;
  print(max(X, Y) < min(X, Y) + three ? Yes : No);
  exit(0);
}
